libkbfs/backpressure_disk_limiter.go

```diff
@@ -469,6 +480,52 @@ func (bdl *backpressureDiskLimiter) onBlocksDelete(
 	bdl.fileTracker.onBlocksDelete(blockFiles)
 }
 
+func (bdl *backpressureDiskLimiter) onDiskBlockCacheDelete(
+	ctx context.Context, blockBytes int64) {
+	if blockBytes == 0 {
+		return
+	}
+	bdl.lock.Lock()
+	defer bdl.lock.Unlock()
+	bdl.diskCacheByteTracker.onBlocksDelete(blockBytes)
+}
+
+func (bdl *backpressureDiskLimiter) beforeDiskBlockCachePut(
+	ctx context.Context, blockBytes int64) (
+	availableBytes int64, err error) {
+	if blockBytes == 0 {
+		// Better to return an error than to panic in ForceAcquire.
+		return 0, errors.New("backpressureDiskLimiter.beforeDiskBlockCachePut" +
+			" called with 0 blockBytes")
+	}
+	bdl.lock.Lock()
+	defer bdl.lock.Unlock()
+	freeBytes, _, err := bdl.freeBytesAndFilesFn()
+	if err != nil {
+		return 0, err
+	}
+
+	bt := bdl.diskCacheByteTracker
+	bt.updateFree(freeBytes)
```

Comment:
Ugh, you're right. I'm not a fan of that solution though. I'd really like to keep the trackers concerns separate, preferably by abstracting out overall usage to an overall tracker.